docs/product-model.md

```diff
@@ -959,35 +991,60 @@ type CalculationPeriodDates:
 
 ### Underlier
```

Comment:
The "Underlier" section will probably need some adjustment depending on how the final model is structured - e.g. a single `Underlier` type?